### PR TITLE
Small adjustments for Path module doc

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR makes some minor adjustments in the Path module documentation.

- Makes module level `const`s `param`
- Split-inits them to "hide" the default values
  - However, adds the default value in documentation to be more expressive about
    them.
- Adjusts basename/dirname documentation to clarify what they do.

These items all came during the module interface review or followups.

Test:
- [x] make docs
- [x] quickstart in `test/library/standard/Path`
- [x] standard
- [x] gasnet

### CODE REVIEW

File: modules/standard/Path.chpl [hunk 1]
```diff
@@ -70,14 +70,30 @@ use SysError, IO;
...
+
+/* 
+   Represents generally the parent directory. On all the platforms that Chapel
+   supports this parameter is set to "..".
+*/
+param parentDir;
+
+/* 
+   Denotes the separator between a directory and its child.  On all the
+   platforms that Chapel supports this parameter is set to "/
```
Comment (lydia-duncan) [inline modules/standard/Path.chpl:88]: Missing a closing `"` :)

**Review by lydia-duncan — APPROVED** [review]
One minor oversight that needs to be fixed, but I don't feel the need to look at it again afterwards.  Thanks!

File: modules/standard/Path.chpl (hunk 1 above)
Comment (e-kayrakli) [reply modules/standard/Path.chpl]: Good catch. Fixed.

**Comment by e-kayrakli** [comment]
I'll merge this if the CI is happy.

**Pull request merged**